orc/src/main/java/org/apache/iceberg/orc/ORCSchemaUtil.java

```diff
@@ -0,0 +1,412 @@
+    final TypeDescription orcType;
+
+    switch (type.typeId()) {
+      case BOOLEAN:
+        orcType = TypeDescription.createBoolean();
+        break;
+      case INTEGER:
+      case TIME:
+        orcType = TypeDescription.createInt();
+        break;
+      case LONG:
+        orcType = TypeDescription.createLong();
+        break;
+      case FLOAT:
+        orcType = TypeDescription.createFloat();
+        break;
+      case DOUBLE:
+        orcType = TypeDescription.createDouble();
+        break;
+      case DATE:
+        orcType = TypeDescription.createDate();
+        break;
+      case TIMESTAMP:
+        orcType = TypeDescription.createTimestamp();
+        break;
+      case STRING:
+        orcType = TypeDescription.createString();
+        break;
+      case UUID:
+      case FIXED:
+      case BINARY:
+        orcType = TypeDescription.createBinary();
+        break;
+      case DECIMAL: {
+        Types.DecimalType decimal = (Types.DecimalType) type;
+        orcType = TypeDescription.createDecimal()
+            .withScale(decimal.scale())
+            .withPrecision(decimal.precision());
+        break;
+      }
+      case STRUCT: {
+        orcType = TypeDescription.createStruct();
+        for (Types.NestedField field : type.asStructType().fields()) {
+          TypeDescription childType = toOrc(field.fieldId(), field.type(), field.isRequired());
+          orcType.addField(field.name(), childType);
+        }
+        break;
+      }
+      case LIST: {
+        Types.ListType list = (Types.ListType) type;
+        TypeDescription elementType = toOrc(list.elementId(), list.elementType(),
+            list.isElementRequired());
+        orcType = TypeDescription.createList(elementType);
+        break;
+      }
+      case MAP: {
+        Types.MapType map = (Types.MapType) type;
+        TypeDescription keyType = toOrc(map.keyId(), map.keyType(), true);
+        TypeDescription valueType = toOrc(map.valueId(), map.valueType(), map.isValueRequired());
+        orcType = TypeDescription.createMap(keyType, valueType);
+        break;
+      }
+      default:
+        throw new IllegalArgumentException("Unhandled type " + type.typeId());
+    }
+
+    // Set Iceberg column attributes for mapping
+    orcType.setAttribute(ICEBERG_ID_ATTRIBUTE, String.valueOf(fieldId));
+    orcType.setAttribute(ICEBERG_REQUIRED_ATTRIBUTE, String.valueOf(isRequired));
+    return orcType;
+  }
+
+  /**
+   * Converts an Iceberg schema to a corresponding ORC schema within the context of an existing
+   * ORC file schema. This method also handles schema evolution from the original ORC file schema
+   * to the given Iceberg schema.
+   *
+   * @param schema an Iceberg schema
+   * @param originalOrcSchema an existing ORC file schema
+   * @return the resulting ORC schema
+   */
+  static TypeDescription toOrc(Schema schema, TypeDescription originalOrcSchema) {
```

Comment:
This is really building a projection of the ORC schema for reading, right? I think a different name would be better because this is not really converting. It is building an ORC projection schema based on the given Iceberg projection schema. Maybe `buildORCProjection` like Avro's `buildAvroProjection`.